### PR TITLE
E2E: add test case L00008 && A00009 for spiderpool

### PR DESCRIPTION
**Pull request by ty-dc**
Signed-off-by: ty-dc [tao.yang@daocloud.io](mailto:tao.yang@daocloud.io)

What type of PR is this?
/issue-bug pr/release/none-required

What this PR does / why we need it:
- add e2e case L00008 for spiderpool
   - Successfully restarted statefulSet/pod with matching podSelector, ip remains the same
- add e2e case A00009 for spiderpool
   - Modify the annotated IPPool for a specified Deployment pod; Modify the annotated IPPool for a specified StatefulSet pod

Which issue(s) this PR fixes:
#684

### CODE REVIEW

File: test/e2e/common/spiderpool.go [hunk 1]
```diff
@@ -615,3 +615,22 @@ func WaitIppoolStatusConditionByAllocatedIPs(ctx context.Context, f *frame.Frame
...
+func WaitWorkloadDeleteUntilFinish(ctx context.Context, f *frame.Framework, namespace, name string) error {
+	if f == nil || namespace == "" || name == "" {
+		return frame.ErrWrongInput
+	}
+	for {
+		select {
+		case <-ctx.Done():
+			return errors.New("time out to wait Workload delete until finish")
+		default:
+			workload := GetWorkloadByName(f, namespace, name)
```
Comment (weizhoublue) [inline test/e2e/common/spiderpool.go:629]: GetWorkloadByName 这个 api 是谁写的，得修下，连 err 返回都没有 
f.GetResource 有 err 就表示  SpiderEndpoint  nil ？ 如果是网络不通呢？
```
func GetWorkloadByName(f *frame.Framework, namespace, name string) *v1.SpiderEndpoint {
     ...
	e := f.GetResource(v, existing)
	if e != nil {
		return nil
	}
	return existing
}
```

File: test/e2e/affinity/affinity_test.go [hunk 2]
```diff
@@ -306,6 +306,144 @@ var _ = Describe("test Affinity", Label("affinity"), func() {
...
+		var newPodList *corev1.PodList
+		const stsOriginialNum = int(1)
+
+		BeforeEach(func() {
+			// Init test info and create namespace
+			nic = "eth0"
+			statefulSetName = "sts" + tools.RandomName()
+			namespace = "ns" + tools.RandomName()
+			GinkgoWriter.Printf("create namespace %v \n", namespace)
+			err := frame.CreateNamespaceUntilDefaultServiceAccountReady(namespace, time.Second*10)
```
Comment (weizhoublue) [inline test/e2e/affinity/affinity_test.go:320]: 每个 context 里的  beforeeach 里 都有 frame.CreateNamespaceUntilDefaultServiceAccountReady 

为什么不 抽象一个最外层的 beforeeach ， 降低代码量，避免相互copy

File: test/e2e/affinity/affinity_test.go [hunk 3]
```diff
@@ -306,6 +306,144 @@ var _ = Describe("test Affinity", Label("affinity"), func() {
...
+					Expect(len(newPodList.Items)).Should(Equal(stsOriginialNum))
+					for _, pod := range newPodList.Items {
+						for _, c := range pod.Status.ContainerStatuses {
+							if _, ok := containerIdMap[c.ContainerID]; ok {
+								time.Sleep(time.Second)
+								continue LOOP
+							}
+							break LOOP
+						}
+					}
```
Comment (weizhoublue) [inline test/e2e/affinity/affinity_test.go:404]: 这里 是否抽象一个 api  ，restartStatefulsetUntilReady ?

File: test/e2e/affinity/affinity_test.go [hunk 4]
```diff
@@ -306,6 +306,144 @@ var _ = Describe("test Affinity", Label("affinity"), func() {
...
+				}
+			}
+			for _, pod := range newPodList.Items {
+				// IP remains the same
+				if frame.Info.IpV4Enabled {
+					podIPv4, ok := tools.CheckPodIpv4IPReady(&pod)
+					Expect(ok).NotTo(BeFalse(), "failed to get ipv4 ip")
+					Expect(podIPv4).NotTo(BeEmpty(), "podIPv4 is a empty string")
+					d, ok := ipMap[podIPv4]
+					Expect(ok).To(BeTrue(), "Failed to get the same Pod IPv4 IP")
```
Comment (yulng) [inline test/e2e/affinity/affinity_test.go:414]: Failed to get the same IPv4 address of Pod

File: test/e2e/affinity/affinity_test.go [hunk 5]
```diff
@@ -306,6 +306,144 @@ var _ = Describe("test Affinity", Label("affinity"), func() {
...
+					GinkgoWriter.Printf("Pod %v IP %v remains the same \n", d, podIPv4)
+				}
+				if frame.Info.IpV6Enabled {
+					podIPv6, ok := tools.CheckPodIpv6IPReady(&pod)
+					Expect(ok).NotTo(BeFalse(), "failed to get ipv6 ip")
+					Expect(podIPv6).NotTo(BeEmpty(), "podIPv6 is a empty string")
+					d, ok := ipMap[podIPv6]
+					Expect(ok).To(BeTrue(), "Failed to get the same Pod IPv6 IP")
```
Comment (yulng) [inline test/e2e/affinity/affinity_test.go:422]: Failed to get the same IPv6 address of Pod

File: test/e2e/affinity/affinity_test.go (hunk 2 above)
Comment (ty-dc) [reply test/e2e/affinity/affinity_test.go]: 已修改

File: test/e2e/affinity/affinity_test.go (hunk 5 above)
Comment (ty-dc) [reply test/e2e/affinity/affinity_test.go]: 已修改

**Review by bzsuni — APPROVED** [review]

**Pull request merged**